src/super_gradients/common/environment/monitoring/data_models.py

```diff
@@ -0,0 +1,55 @@
+import dataclasses
+from functools import partial
+from typing import Callable, List, Iterator
+
+from super_gradients.common.environment.monitoring.gpu import init_nvidia_management_lib, count_gpus
+
+
+@dataclasses.dataclass
+class StatAggregator:
+    """Accumulate statistics samples and aggregates them.
+
+    :param name:            Name of the statistic
+    :param sampling_fn:     How the statistic is sampled
+    :param aggregate_fn:    How the statistic samples are aggregated
+    """
+
+    name: str
+    sampling_fn: Callable
+    aggregate_fn: Callable
+    _samples: List = dataclasses.field(default_factory=list)
+
+    def sample(self):
+        self._samples.append(self.sampling_fn())
+
+    def aggregate(self) -> float:
+        value = self.aggregate_fn(self._samples)
+        self._samples = []
```

Comment:
please move this into a separate function `reset()`.
It took me a while to understand you are resetting on every `aggregate()` call

Reply:
Added in 4be886f65b51c41b1b20074c04d5d19b8a87ac1a